app/models/manager_refresh/inventory_collection_default.rb

```diff
@@ -1,2 +1,98 @@
 class ManagerRefresh::InventoryCollectionDefault
+  class << self
+    def vms(extra_attributes = {})
```

Comment:
Does it matter for these consolidated methods that they combine the attributes of both infra and cloud?  I guess the real question is, what happens in infra when it sees attributes that previously only existed for cloud refreshes for vms?

An example would be `attributes_blacklist`.  Does infra care about this new attribute?

Reply:
Yes, I think the important part here is that we've extracted the same behavior to one place. And we can overwrite that behavior in providers repos, if needed.

Reply:
Btw. the blacklist behavior is being used in both Infra and Cloud

Reply:
Right so I think these consolidated methods should only contain what is strictly common between infra and cloud, anything specific should be added by the sub-class.  And yeah like @Ladas said `:attribute_blacklist` is not a cloud specific property.

Reply:
Maybe I'm reading this wrong... but the [`InfraManager` `vms` method](https://github.com/ManageIQ/manageiq/pull/15941/files#diff-282a4caf529c48df7d0ed652af48a4c8L3) looked like:

```ruby
def vms(extra_attributes = {})
  attributes = {
    :model_class    => ::ManageIQ::Providers::InfraManager::Vm,
    :association    => :vms,
    :builder_params => {
      :ems_id => ->(persister) { persister.manager.id },
    },
  }

  attributes.merge!(extra_attributes)
end
```

And, the [`CloudManager` `vms` method](https://github.com/ManageIQ/manageiq/pull/15941/files#diff-ec6abf1e9f2c21d002c092c65a3383e5L3) looked like:

```ruby
def vms(extra_attributes = {})
  attributes = {
    :model_class            => ::ManageIQ::Providers::CloudManager::Vm,
    :association            => :vms,
    :delete_method          => :disconnect_inv,
    :attributes_blacklist   => [:genealogy_parent],
    :use_ar_object          => true, # Because of raw_power_state setter
    # TODO(lsmola) can't do batch strategy for vms because of key_pairs relation
    :saver_strategy         => :default,
    :batch_extra_attributes => [:power_state, :state_changed_on, :previous_state],
    :builder_params         => {
      :ems_id   => ->(persister) { persister.manager.id },
      :name     => "unknown",
      :location => "unknown",
    }
  }

  attributes.merge!(extra_attributes)
end
```

So, the consolidate implementation looks like it took the previous `CloudManager` implementation as the default.  Does that mean that any `InfraManager` will have to subtract things?  Or, that it simply won't care that new things will suddenly exist in the `attributes` hash?

Again, maybe I'm mis-reading this.  But, it looks like `InfraManager` will get new entries in the `attributes` hash.

Reply:
Nope, CloudManager is just father in Targeted refresh and better tested. All of those options should be shared

Reply:
Yeah those `:builder_params` are needed for Infra Vms as well in order to do batch saving, instead of adding them I thought it would be better to unify the collections.